Zilon.Core/Zilon.Bot.Players/BotActorTaskSourceBase.cs

```diff
@@ -3,49 +3,37 @@
 using System.Threading.Tasks;
 
 using Zilon.Bot.Sdk;
-using Zilon.Core.Persons;
-using Zilon.Core.Players;
 using Zilon.Core.Tactics;
 using Zilon.Core.Tactics.Behaviour;
 
 namespace Zilon.Bot.Players
 {
-    public abstract class BotActorTaskSourceBase : ISectorActorTaskSource
+    public abstract class BotActorTaskSourceBase<TContext> : ISectorActorTaskSource<TContext> where TContext: class, ISectorTaskSourceContext
     {
-        private readonly IPlayer _player;
-
+        //TODO Есть риск утечки.
+        // Актёры могут быть удалены, но информация о них будет храниться здесь, предотвращая чистку.
+        // Рассмотреть варианты:
+        // 1. Периодическая чистка внутри самого источника команд. Тогда нужна зависимость от менеджера актёров.
+        // 2. В интерфейс добавить метод для обработки удаления актёра.
         private readonly Dictionary<IActor, ILogicStrategy> _actorStrategies;
 
-        public BotActorTaskSourceBase(IPlayer player)
+        public BotActorTaskSourceBase()
```

Comment:
Codacy found an issue: [Change the visibility of this constructor to 'protected'.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=6105716)